src/platform/platform_winuser.c

```diff
@@ -104,30 +103,29 @@ CxPlatProcessorInfoInit(
             "Allocation of '%s' failed. (%llu bytes)",
             "CxPlatProcessorInfo",
             ActiveProcessorCount * sizeof(CXPLAT_PROCESSOR_INFO));
+        Status = QUIC_STATUS_OUT_OF_MEMORY;
         goto Error;
     }
 
-    if (!CxPlatProcessorGroupInfo(
+    Status =
+        CxPlatProcessorGroupInfo(
             RelationAll,
             (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX*)&Buffer,
-            &BufferLength)) {
+            &BufferLength);
+    if (QUIC_FAILED(Status)) {
         goto Error;
     }
 
     Offset = 0;
     while (Offset < BufferLength) {
         PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX Info =
             (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX)(Buffer + Offset);
-        if (Info->Relationship == RelationNumaNode) {
-            if (Info->NumaNode.NodeNumber + 1 > NumaNodeCount) {
-                NumaNodeCount = Info->NumaNode.NodeNumber + 1;
-            }
-        } else if (Info->Relationship == RelationGroup) {
-            if (ProcessorGroupCount == 0) {
-                CXPLAT_DBG_ASSERT(Info->Group.ActiveGroupCount != 0);
-                ProcessorGroupCount = Info->Group.ActiveGroupCount;
-                ProcessorsPerGroup = Info->Group.GroupInfo[0].ActiveProcessorCount;
-            }
+        if (Info->Relationship == RelationGroup) {
+            CXPLAT_DBG_ASSERT(ProcessorGroupCount == 0);
+            CXPLAT_DBG_ASSERT(Info->Group.ActiveGroupCount != 0);
+            ProcessorGroupCount = Info->Group.ActiveGroupCount;
+            ProcessorsPerGroup = Info->Group.GroupInfo[0].ActiveProcessorCount;
```

Comment:
Is it safe to make the assumption that `ProcessorsPerGroup` is identical in each group?

Reply:
I _thought_ so, but https://learn.microsoft.com/en-us/windows/win32/procthread/processor-groups makes no mention of that assumption. The closest I can find is:
> The operating system minimizes the number of groups in a system. For example, a system with 128 logical processors would have two processor groups with 64 processors in each group, not four groups with 32 logical processors in each group.

Reply:
Latest commits should handle any scenario.

Reply:
Looks good.